web/src/app/schedules/ScheduleShiftList.js

```diff
@@ -266,12 +267,23 @@ export default class ScheduleShiftList extends React.PureComponent {
       <TextField
         fullWidth
         label='Time Limit (days)'
-        value={Duration.fromISO(this.props.duration).as('days')}
+        value={
+          this.state.isClear
+            ? ''
+            : Duration.fromISO(this.props.duration).as('days')
+        }
         disabled={this.props.activeOnly}
         max={30}
         min={1}
         type='number'
+        onBlur={() => {
+          this.setState({ isClear: false })
+        }}
         onChange={(e) => {
+          this.setState({ isClear: e.target.value === '' })
+          if (parseInt(e.target.value) < 0) {
+            return
+          }
```

Comment:
I think this is also handled by the `clamp` function below since -1, 0, etc.. is going to be less than 1 (as written).

The only thing it won't handle though is `NaN`. The logic you had before `if (parseInt(...) > 0) { handleSetDuration() }` would implicitly check that it was a valid number as well as `> 0`. 

Basically in javascript the logical inverse of `parseInt(x) > 0` would be either `!parseInt(x) <= 0` or `Number.isNaN(parseInt(x)) || parseInt(x) <= 0`. But since we're "clamping" the value already, it's probably enough to do the `isNaN` check here to avoid the redundant check.

```suggestion
          if (Number.isNaN(parseInt(e.target.value, 10))) {
            return
          }
```

This should ensure invalid values/states are ignored.

Reply:
thanks for the feedback. The solution proposed is addressed and implemented.